modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -164,6 +164,10 @@ public GridDhtPartitionDemander(CacheGroupContext grp) {
         mreg.register("RebalancingPartitionsLeft", () -> rebalanceFut.partitionsLeft.get(),
             "The number of cache group partitions left to be rebalanced.");
 
+        mreg.register("RebalancingPartitionsTotal",
+            () -> rebalanceFut.rebalancingParts.values().stream().mapToInt(Set::size).sum(),
```

Comment:
This value is a constant and should be precomputed on future creation to make metrics overhead as low as possible.